uaclient/entitlements/base.py

```diff
@@ -273,7 +278,26 @@ def process_contract_deltas(
 
         return False
 
-    @abc.abstractmethod
     def operational_status(self) -> 'Tuple[str, str]':
         """Return whether entitlement is ACTIVE, INACTIVE or UNAVAILABLE"""
+        applicability, details = self.applicability_status()
+        if applicability != ApplicabilityStatus.APPLICABLE:
+            return status.INAPPLICABLE, details
+        entitlement_cfg = self.cfg.entitlements.get(self.name)
+        if not entitlement_cfg:
+            return status.INAPPLICABLE, '%s is not entitled' % self.title
+        elif entitlement_cfg['entitlement'].get('entitled', False) is False:
+            return status.INAPPLICABLE, '%s is not entitled' % self.title
+
+        application_status, explanation = self.application_status()
+        return application_status.to_operational_status(), explanation
+
+    @abc.abstractmethod
+    def application_status(self) -> 'Tuple[status.ApplicationStatus, str]':
+        """
+        The current status of application of this entitlement
+
+        :return:
+            A tuple of (ApplicationStatus, human-friendly reason)
```

Comment:
human-friendly reason... are you saying humans are inherently unreasonable?

Reply:
Are you saying they aren't?! :-P